config/service.go

```diff
@@ -53,13 +53,16 @@ type ServiceTelemetryLogs struct {
 	Development bool `mapstructure:"development"`
 
 	// Encoding sets the logger's encoding.
-	// Valid values are "json" and "console".
+	// Valid values are "json", "console" and "custom".
 	Encoding string `mapstructure:"encoding"`
+
+	/// If encoding is set to custom, uses this setting
+	CustomEncoding string `mapstructure:"custom_encoding"`
```

Comment:
I am extremely confused why not writing "xml" directly in the "encoding" field, because it looks like this is what is going to happen in logger.go at line 27.

Reply:
I used this approach to not break current validation. Of course writing "xml" (as encoder name) directly to this field could be clearer. 
I'm open for suggestion which approach is better.

Reply:
You are not breaking current validation, it is allowed to support more encodings, it is considered breaking change removing them.

Reply:
Should we add "xml" to the allowed types for encoding below?  This seems to enable bypassing the validation entirely by setting `encoding` to "custom" and putting anything in `custom_encoding`.

Reply:
@bogdandrutu If I ran collector with this config:

```yml
receivers:
  otlp:
    protocols:
      http:
exporters:
  logging:
service:
  pipelines:
    traces:
      receivers: [otlp]
      exporters: [logging]
  telemetry:
    logs:
      encoding: xml
```
I see following error message:

```
Error: invalid configuration: service telemetry logs invalid encoding: "xml", valid values are "json" and "console"
2021/12/13 14:06:33 collector server run finished with error: invalid configuration: service telemetry logs invalid encoding: "xml", valid values are "json" and "console"
```

My PR relaxing a bit this validation by adding possibility to use custom encoding with the separate field. 

`xml` is used only for testing purposes, it can be any other name specified by user in registration in zap.

Reply:
I am struggling to understand this... sorry... But just adding an extra field that has the same value is extremely confusing... Not sure what you want to do but this will not be accepted as is. See previous comments from me and @Aneurysm9

Reply:
Once again I will try explain my motivations:

- I created custom otel-collector distribution with some custom processing and extensions
- In single extension I registering custom encoder to `zap`
- Then I want use this extension for encoding logs from `otel-collector`. I can't do that because `otel-collector` doesn't allow set custom encoding name. Only `console` or `json` setting is allowed.
- @Aneurysm9 pointed out to add `xml` as another encoding option, but `xml` is only example name of encoder. 
- Finally, I created this PR to relax encoding name validation.

So, with your suggestions we have two options here: 
- my PR with custom field (which is not great, but it's not breaking anything)
 - removing validation for encoding field

Please, share your thoughts about that approaches and I prepare final PR.

Reply:
> I can't do that because otel-collector doesn't allow set custom encoding name. Only console or json setting is allowed.

Indeed "removing validation for encoding field" is probably better to not confuse people. Leave a comment that by default only these values are supported, but in a custom distros this may support more encodings. Then we will not fail as fast (in the parsing) but during start.

To help our users we can check for "console" and "json" and print a logging message somewhere that "you are using a non standard encoding".

Reply:
Thanks for clarification. I will update PR soon

Reply:
PR updated